csharp/src/Ice/SocketStream.cs

```diff
@@ -52,7 +52,9 @@ public long Id
         /// <summary>The Reset event is triggered when a reset frame is received.</summary>
         internal event Action<long>? Reset;
 
-        private protected bool IsStarted => _id != -1;
+        /// <summary>Returns true if the stream ID is assigned</summary>
+        internal bool IsStarted => _id != -1;
```

Comment:
`_id` is not volatile or protected by a mutex - would be nice to add a comment explaining why.

Reply:
Added the following:
```
        // Depending on the stream implementation, the _id can be assigned on construction or only once SendAsync
        // is called. Once it's assigned, it's immutable. The specialization of the stream is responsible for not
        // accessing this data member concurrently when it's not safe.
```